node/v2/handler.js

```diff
@@ -630,36 +630,33 @@ TChannelV2Handler.prototype.verifyCallRequestFrame =
 function verifyCallRequestFrame(req, args) {
     var self = this;
 
-    var message;
-    if (self.requireAs) {
-        message = 'Expected the "as" transport header to be set for request\n' +
-            'Got request for ' + req.serviceName + ' ' + req.endpoint +
-            ' without as header';
-
-        assert(req.headers && req.headers.as, message);
-    } else if (!req.headers || !req.headers.as) {
-        self.logger.error('Expected "as" header to be set for request', {
-            arg1: req.endpoint,
-            callerName: req.headers && req.headers.cn,
-            remoteName: self.remoteName,
-            serviceName: req.serviceName,
-            socketRemoteAddr: self.connection.socketRemoteAddr
-        });
+    if (!req.headers || !req.headers.as) {
+        if (self.requireAs) {
```

Comment:
Did we not implement a version of this that could be safely employed at Hyperbahn ingress?